### PR TITLE
Add i.MX RT I2C support

### PR DESCRIPTION
**Pull request by MaureenHelm**
Introduces a new mcux lpi2c shim driver for the i.MX RT series. This is the first step towards enabling the capacitive touch sensor on the display.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
doc changes LGTM

**Comment by MaureenHelm** [comment]
recheck

File: boards/arm/mimxrt1050_evk/dts_fixup.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
```
Comment (galak) [inline boards/arm/mimxrt1050_evk/dts_fixup.h]: Don't need fixup for DT_FXOS8700_ anymore.

File: drivers/i2c/Kconfig [hunk 2]
```diff
@@ -267,6 +267,13 @@ config I2C_MCUX
 	help
 	  Enable the mcux I2C driver.
 
+config I2C_MCUX_LPI2C
+	bool "MCUX LPI2C driver"
```
Comment (galak) [inline drivers/i2c/Kconfig]: Does it make sense to have this be something like:

`bool "MCUX LPI2C driver (used on i.MX-RT)"`

**Review by galak — CHANGES_REQUESTED** [review]
Some minor comments, but generally looks good.

**Review by galak — COMMENTED** [review]
Do you want to add the board to samples/sensor/fxos8700/sample.yaml?

Might now be worth changing this sample to be like samples/sensor/hts221/sample.yaml and have it be 'depends_on'.

File: drivers/i2c/Kconfig (hunk 2 above)
Comment (MaureenHelm) [reply drivers/i2c/Kconfig]: No, because there are some newer kinetis parts with this IP

**Comment by MaureenHelm** [comment]
> Do you want to add the board to samples/sensor/fxos8700/sample.yaml?

Not yet, because the sample currently requires the sensor interrupts.

**Comment by MaureenHelm** [comment]
> Ok, I think just removing the dts_fixup.h modification and it looks good.

Done

**Review by galak — APPROVED** [review]

**Review by jhqian — APPROVED** [review]

**Review by agansari — APPROVED** [review]

**Pull request merged**